erpnext_germany/erpnext_germany/doctype/vat_id_check/vat_id_check.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2023, ALYF GmbH and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe.model.document import Document
+from erpnext_germany.utils.eu_vat import check_vat_approx, parse_vat_id
+
+from tenacity import RetryError
+
+
+class VATIDCheck(Document):
+	def before_insert(self):
+		if self.requester_vat_id:
+			requester_country_code, requester_vat_number = parse_vat_id(
+				self.requester_vat_id
+			)
+			self.requester_vat_id = f"{requester_country_code}{requester_vat_number}"
+
+		country_code, vat_number = parse_vat_id(self.customer_vat_id)
+		self.customer_vat_id = f"{country_code}{vat_number}"
+
+	def after_insert(self):
+		frappe.enqueue(
+			run_check,
+			doc=self,
+			queue="long",
+			now=frappe.conf.developer_mode or frappe.flags.in_test,
+		)
+
+
+def run_check(doc: VATIDCheck):
+	doc.db_set("status", "Running", notify=True)
+	requester_country_code, requester_vat_number = None, None
+	if doc.requester_vat_id:
+		requester_country_code, requester_vat_number = parse_vat_id(doc.requester_vat_id)
+
+	country_code, vat_number = parse_vat_id(doc.customer_vat_id)
+	try:
+		result = check_vat_approx(
+			country_code=country_code,
+			vat_number=vat_number,
+			trader_name=doc.trader_name,
+			trader_street=doc.trader_street,
+			trader_postcode=doc.trader_postcode,
+			trader_city=doc.trader_city,
+			requester_country_code=requester_country_code,
+			requester_vat_number=requester_vat_number
+		)
+	except RetryError:
+		doc.db_set("status", "Service Unavailable", notify=True)
+		return
+	except Exception as e:
+		if e.message.upper() == "INVALID_INPUT":
+			doc.db_set({"status": "Invalid Input", "is_valid": False}, notify=True)
+			return
+		raise e
+
+	doc.db_set(
+		{
+			"status": "Completed",
+			"is_valid": result.valid,
+			"trader_name_match": is_match(result.traderNameMatch, result.traderName, doc.trader_name, exact=True),
+			"trader_street_match": is_match(result.traderStreetMatch, result.traderAddress, doc.trader_street),
+			"trader_postcode_match": is_match(result.traderPostcodeMatch, result.traderAddress, doc.trader_postcode),
+			"trader_city_match": is_match(result.traderCityMatch, result.traderAddress, doc.trader_city),
+			"request_id": result.requestIdentifier,
+			"actual_trader_name": result.traderName,
+			"actual_trader_address": result.traderAddress,
+		},
+		notify=True,
+	)
+
+
+def is_match(match, theirs, ours, exact=False):
+	if match is not None:
+		return bool(match)
+
+	if exact:
+		return ours.lower() == theirs.lower()
+
+	return ours.lower() in theirs.lower()
```

Comment:
ours and theirs could be None in case of an invalid VAT ID
```py
File .../erpnext_germany/erpnext_germany/doctype/vat_id_check/vat_id_check.py:66, in run_check(doc)
     58 		return
     59 	raise e
     61 doc.db_set(
     62 	{
     63 		"status": "Completed",
     64 		"is_valid": result.valid,
     65 		"trader_name_match": is_match(result.traderNameMatch, result.traderName, doc.trader_name, exact=True),
---> 66 		"trader_street_match": is_match(result.traderStreetMatch, result.traderAddress, doc.trader_street),
     67 		"trader_postcode_match": is_match(result.traderPostcodeMatch, result.traderAddress, doc.trader_postcode),
     68 		"trader_city_match": is_match(result.traderCityMatch, result.traderAddress, doc.trader_city),
     69 		"request_id": result.requestIdentifier,
     70 		"actual_trader_name": result.traderName,
     71 		"actual_trader_address": result.traderAddress,
     72 	},
     73 	notify=True,
     74 )

File .../erpnext_germany/doctype/vat_id_check/vat_id_check.py:84, in is_match(match, theirs, ours, exact)
     81 if exact:
     82 	return ours.lower() == theirs.lower()
---> 84 return ours.lower() in theirs.lower()

AttributeError: 'NoneType' object has no attribute 'lower'
```

Result of `VAT ID: DE123456789`
```py
{
    'countryCode': 'DE',
    'vatNumber': '123456789',
    'requestDate': datetime.date(2023, 7, 14),
    'valid': False,
    'traderName': 'Marica',
    'traderCompanyType': None,
    'traderAddress': None,
    'traderStreet': '14, Peterssteinweg',
    'traderPostcode': '04107',
    'traderCity': 'Leipzig',
    'traderNameMatch': None,
    'traderCompanyTypeMatch': None,
    'traderStreetMatch': None,
    'traderPostcodeMatch': None,
    'traderCityMatch': None,
    'requestIdentifier': None
}
```